### ISSUE TITLE
The use-os returns wrong OS on iPad iOS 13

### ISSUE THREAD

**Issue by mattp0123** [mantinedev/mantine#1082]
### What package has an issue

@mantine/hooks

### Describe the bug

As titled. The `window.navigator.platform` on iPad iOS 13 is `MacIntel`.

### In which browser did the problem occur

Safari 13 on iPad

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

### Possible fix

```TypeScript
type OS = 'windows' | 'mac' | 'linux' | 'ios' | 'android'

function getOS(): OS {
  const { platform, maxTouchPoints = 0 } = window.navigator;
  
  if (platform === 'MacIntel' && maxTouchPoints > 1) {
    return 'ios'
  }

  // Rest conditions
}
```

**Comment by karan-bhatia22** [comment]
You sure about this? Can you provide a screenshot?

**Comment by mattp0123** [comment]
@karan-bhatia22
https://stackoverflow.com/questions/57765958/how-to-detect-ipad-and-ipad-os-version-in-ios-13-and-up

**Comment by karan-bhatia22** [comment]
It seems like `navigator.platform` is deprecated. Using regex on `navigator.userAgent` would be a workaround.

**Comment by rtivital** [comment]
Fixed in 4.0.9 release
